### DISCUSSION TITLE
Remove/revise dependencies

### DISCUSSION THREAD

**Discussion by lauraguzmanrincon** [epiverse-trace/research-compendium#5] [Q&A]
Sometimes I install more dependencies than actually needed.

Is there a way to revise the list of dependencies and find the ones that are not required anymore? To keep the project as simple as possible?

**Comment by avallecam** [comment]
Hola Laura,

Thank you for your question. I explored the Reference guide of the [{renv}](https://rstudio.github.io/renv/reference/index.html) package and found two useful functions for this:
- [`renv::dependencies()`](https://rstudio.github.io/renv/reference/dependencies.html) generates a table of all the _direct_ dependencies and the source file inside the project. Really informative output to have a summarized overview. I run `View(renv::dependencies())` to look at it in a separate View tab.
- [`renv::remove()`](https://rstudio.github.io/renv/reference/remove.html) can remove any _direct_ dependency from this list.

What are _direct_ dependencies? These are packages that are directly or explicitly used in a project file. The _indirect_ dependencies refer to all the dependencies that _direct_ package dependencies need to run. [`renv::diagnostics()`](https://rstudio.github.io/renv/reference/diagnostics.html) allow to look at both in more detail. In that output, packages like `{rcompendium}` and `{cffr}` are registered as `<NA>` because they were used to create something for the project but not directly written in a file.
